hibernate-core/src/main/java/org/hibernate/engine/spi/SessionDelegatorBaseImpl.java

```diff
@@ -1163,7 +1163,7 @@ public Integer getJdbcBatchSize() {
 	}
 
 	@Override
-	public void setJdbcBatchSize(int jdbcBatchSize) {
+	public void setJdbcBatchSize(Integer jdbcBatchSize) {
```

Comment:
I think optional is great when used for methods like getSingleResult or for chaining method calls. 

While for `getJdbcBatchSize` it might make sense, even if the end-user might not need to query this setting too often, for `setJdbcBatchSize`, `Optional` might just be a little bit too verbose.

In this particular use case, if we change the signature to `Optional` the call would look like this:

    session.setJdbcBatchSize(Optional.of(10));
    session.setJdbcBatchSize(Optional.ofNullable(null));

For the case when we want to reset the Session-level setting, we don't really gain anything from the new syntax. 

If you want to be more strict about this parameter type, we could just add a new method `resetJdbcBatchSize` and rollback the `setBatchSize` method to take an `int` instead of an `Integer`.